### ISSUE TITLE
Color output of compiler warnings and errors show incorrectly in Output panel and are not parsed for Problems panel

### ISSUE THREAD

**Issue by reedhedges** [microsoft/vscode-cmake-tools#2278]
Hello, I'm not sure if this is an issue with CMake tools, or something else (vscode's Terminal? I've tried both "internal" and "external" settings), so if anyone has any suggestions  for investigating further to narrow down what's actually happening, it would be appreciated.

When I build, compiler errors and warnings are not being displayed correctly in the Output pane, and Problems doesn't pick them up at all.   The terminal escape codes for color are not working correctly (see output below).

I am on Linux, and have set "Unix Makefiles" as preferred generator.  I normally leave the compiler setting unspecified for cmake tools, but have also set gcc and clang versions (kits) with same results.  I have the standard "C/C++" extension installed, along with "C/C++ Themes", "CMake" some others, but disabled all of those with the same results. 

```
[main] Building folder: cmake_example 
[main] Configuring folder: cmake_example 
[proc] Executing command: /usr/bin/cmake --no-warn-unused-cli -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_BUILD_TYPE:STRING=Debug -H/home/reed/examples-tutorials-tests/cmake_example -B/home/reed/examples-tutorials-tests/cmake_example/build -G "Unix Makefiles"
[cmake] Not searching for unused variables given on the command line.
[cmake] -- The CXX compiler identification is GNU 10.3.0
[cmake] -- Detecting CXX compiler ABI info
[cmake] -- Detecting CXX compiler ABI info - done
[cmake] -- Check for working CXX compiler: /usr/bin/c++ - skipped
[cmake] -- Detecting CXX compile features
[cmake] -- Detecting CXX compile features - done
[cmake] -- ccache found and enabled
[cmake] -- Configuring done
[cmake] -- Generating done
[cmake] -- Build files have been written to: /home/reed/examples-tutorials-tests/cmake_example/build
[cmakefileapi-driver] This version of CMake does not support the "toolchains" object kind. Compiler paths will be determined by reading CMakeCache.txt.
[build] Starting build
[proc] Executing command: /usr/bin/cmake --build /home/reed/examples-tutorials-tests/cmake_example/build --config Debug --target all -j 6 --
[build] Scanning dependencies of target example
[build] [ 50%] Building CXX object CMakeFiles/example.dir/example.cc.o
[build] [01m[K/home/reed/examples-tutorials-tests/cmake_example/example.cc:[m[K In function ‘[01m[Kint main()[m[K’:
[build] [01m[K/home/reed/examples-tutorials-tests/cmake_example/example.cc:15:6:[m[K [01;35m[Kwarning: [m[Kignoring return value of ‘[01m[Kint foo(char, int)[m[K’, declared with attribute ‘[01m[Knodiscard[m[K’ [[01;35m[K-Wunused-result[m[K]
[build]    15 |   [01;35m[Kfoo(x, c)[m[K;
[build]       |   [01;35m[K~~~^~~~~~[m[K
[build] [01m[K/home/reed/examples-tutorials-tests/cmake_example/example.cc:6:19:[m[K [01;36m[Knote: [m[Kdeclared here
[build]     6 | [[nodiscard]] int [01;36m[Kfoo[m[K(char c, int x)
[build]       |                   [01;36m[K^~~[m[K
[build] [01m[K/home/reed/examples-tutorials-tests/cmake_example/example.cc:16:3:[m[K [01;31m[Kerror: [m[K‘[01m[Kbar[m[K’ was not declared in this scope
[build]    16 |   [01;31m[Kbar[m[K(x);
[build]       |   [01;31m[K^~~[m[K
[build] [01m[K/home/reed/examples-tutorials-tests/cmake_example/example.cc:17:10:[m[K [01;31m[Kerror: [m[Kinvalid conversion from ‘[01m[Kconst char*[m[K’ to ‘[01m[Kint[m[K’ [[01;31m[K-fpermissive[m[K]
[build]    17 |   return [01;31m[K"hello world"[m[K;
[build]       |          [01;31m[K^~~~~~~~~~~~~[m[K
[build]       |          [01;31m[K|[m[K
[build]       |          [01;31m[Kconst char*[m[K
[build] gmake[2]: *** [CMakeFiles/example.dir/build.make:82: CMakeFiles/example.dir/example.cc.o] Error 1
[build] gmake[1]: *** [CMakeFiles/Makefile2:95: CMakeFiles/example.dir/all] Error 2
[build] gmake: *** [Makefile:103: all] Error 2
[build] Build finished with exit code 2
```

### CMake Tools Diagnostics

```
{
  "os": "linux",
  "vscodeVersion": "1.62.3",
  "cmtVersion": "1.9.2",
  "configurations": [
    {
      "folder": "/home/reed/examples-tutorials-tests/cmake_example",
      "cmakeVersion": "3.18.4",
      "configured": true,
      "generator": "Unix Makefiles",
      "usesPresets": false,
      "compilers": {
        "CXX": "/usr/bin/c++"
      }
    }
  ],
  "cpptoolsIntegration": {
    "isReady": true,
    "hasCodeModel": true,
    "activeBuildType": "Debug",
    "buildTypesSeen": [
      "Debug"
    ],
    "requests": [],
    "responses": [],
    "partialMatches": [],
    "targetCount": 1,
    "executablesCount": 1,
    "librariesCount": 0,
    "targets": [
      {
        "name": "example",
        "type": "EXECUTABLE"
      }
    ]
  },
  "settings": [
    {
      "communicationMode": "automatic",
      "useCMakePresets": "auto",
      "configureOnOpen": true
    }
  ]
}
```
Thanks!

**Comment by reedhedges** [comment]
Update, I've looked more into my CMakeLists.txt and discovered it included some CMake configuration I copied from another source which added some compiler options to configure color output (`-fcolor-diagnostics` for Clang and `-fdiagnostics-color=always` for GCC.)  Removing these fixed the problem.   When run by cmake tools, there was no color output and it was able to parse the output just fine, but when I build from a terminal it does turn on color output.   I have no idea why this third party cmake configuration turned those on, probably to see color output for another IDE.   So I wonder if there's a way to get color output in the "Output" panel in vscode?  I'll updated the title of this issue slightly.
